sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/ThroughputRequestThrottler.java

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.throughputControl;
+
+import com.azure.cosmos.BridgeInternal;
+import com.azure.cosmos.CosmosException;
+import com.azure.cosmos.implementation.HttpConstants;
+import com.azure.cosmos.implementation.RequestRateTooLargeException;
+import com.azure.cosmos.implementation.RxDocumentServiceRequest;
+import com.azure.cosmos.implementation.RxDocumentServiceResponse;
+import com.azure.cosmos.implementation.Utils;
+import com.azure.cosmos.implementation.directconnectivity.StoreResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Mono;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * This is the place where we tracking the RU usage, and make decision whether we should block the request.
+ */
+public class ThroughputRequestThrottler {
+    private static final Logger logger = LoggerFactory.getLogger(ThroughputRequestThrottler.class);
+
+    private final AtomicReference<Double> availableThroughput;
+    private final AtomicReference<Double> scheduledThroughput;
+
+    public ThroughputRequestThrottler(double scheduledThroughput) {
+        this.availableThroughput = new AtomicReference<>(scheduledThroughput);
+        this.scheduledThroughput = new AtomicReference<>(scheduledThroughput);
+    }
+
+    public Mono<Void> renewThroughputUsageCycle(double scheduledThroughput) {
```

Comment:
None of the below operations are reactive, why use Mono<Void> here ?
It will unnecessary complicate the code flow.

Reply:
good catch, will remove Mono in the next iteration.

Reply:
Removed Mono